application/src/main/java/org/opentripplanner/model/Timetable.java

```diff
@@ -296,8 +302,38 @@ public Result<TripTimesPatch, UpdateError> createUpdatedTripTimesFromGTFSRT(
               );
               return Result.failure(new UpdateError(feedScopedTripId, INVALID_DEPARTURE_TIME, i));
             }
-          } else if (delay != null) {
-            newTimes.updateDepartureDelay(i, delay);
+          }
+
+          // propagate arrival and departure times, taking care not to cause negative dwells / hops
```

Comment:
I think you're mixing several features here, aren't you?

Can you please work in small increments? The first PR should only deal with the case where either arrival or departure is missing. In such a case you can use the other one in its place. This shouldn't need any new interpolation logic.

Once we have finished that PR we can look at more complex cases.

Reply:
This is an integral part of the feature. If we just use the other without changing the interpolation, negative times still exist and fails the test I changed.

Reply:
Okay, I don't get it then and I think I need to see drawing then.

Using an arrival/departure fallback is, IMO, still a wise first step and something I would like to see first, even if it doesn't fix all of your problems.

Reply:
The existing interpolation logic isn't what you describe. The existing logic is that if the departure time is missing, it is assumed that the bus will always stop for its original dwell time even if it is late (resulting in negative times down the trip), so what I have written and what you are describing are both changing the logic, which is an integral part of this PR.